commit 3cc1170208f6a0c984e906fc5d531798313d5f40
Author: Irene Wu <irene.w.wu@intel.com>
Date:   Sat Feb 16 02:10:29 2019 -0800

    Resolve github issue #51; Add SONAME in shared IGC & IGA libraries
    
    Change-Id: Id0ce987ded711c85ba9b048b4b55627a7bd8e5c6

diff --git a/IGC/AdaptorOCL/igc.opencl.h.in b/IGC/AdaptorOCL/igc.opencl.h.in
index 263aea6..58d3623 100644
--- a/IGC/AdaptorOCL/igc.opencl.h.in
+++ b/IGC/AdaptorOCL/igc.opencl.h.in
@@ -30,8 +30,8 @@
   #cmakedefine IGC_LIBRARY_NAME     "${IGC_LIBRARY_NAME}32.dll"
   #cmakedefine FCL_LIBRARY_NAME     "${FCL_LIBRARY_NAME}32.dll"
 #else
-  #cmakedefine IGC_LIBRARY_NAME     "lib${IGC_LIBRARY_NAME}.so"
-  #cmakedefine FCL_LIBRARY_NAME     "lib${FCL_LIBRARY_NAME}.so"
+  #cmakedefine IGC_LIBRARY_NAME     "lib${IGC_LIBRARY_NAME}.so.${IGC_API_MAJOR_VERSION}"
+  #cmakedefine FCL_LIBRARY_NAME     "lib${FCL_LIBRARY_NAME}.so.${IGC_API_MAJOR_VERSION}"
 #endif
 
 #endif /* IGC_OPENCL_H */
diff --git a/IGC/CMakeLists.txt b/IGC/CMakeLists.txt
index fd6721d..93d6801 100644
--- a/IGC/CMakeLists.txt
+++ b/IGC/CMakeLists.txt
@@ -53,6 +53,11 @@ endif()
 
 
 
+# Increment IGC_API_MAJOR_VERSION if there is ABI Change.
+set(IGC_API_MAJOR_VERSION 1)
+set(IGC_API_MINOR_VERSION 0)
+set(IGC_API_PATCH_VERSION 0)
+
 # ======================================================================================================
 # ================================================ UTILS ===============================================
 # ======================================================================================================
@@ -3613,6 +3618,9 @@ else()
   igc_arch_get_cpu(_cpuSuffix)
 endif()
 set_target_properties("${IGC_BUILD__PROJ__igc_dll}" PROPERTIES OUTPUT_NAME "${IGC_LIBRARY_NAME}${_cpuSuffix}")
+set_target_properties("${IGC_BUILD__PROJ__igc_dll}" PROPERTIES
+                      VERSION "${IGC_API_MAJOR_VERSION}.${IGC_API_MINOR_VERSION}.${IGC_API_PATCH_VERSION}"
+                      SOVERSION "${IGC_API_MAJOR_VERSION}")
 unset(_cpuSuffix)
 
 #FCL
@@ -3622,6 +3630,9 @@ else()
   igc_arch_get_cpu(_cpuSuffix)
 endif()
 set_target_properties("${IGC_BUILD__PROJ__fcl_dll}" PROPERTIES OUTPUT_NAME "${FCL_LIBRARY_NAME}${_cpuSuffix}")
+set_target_properties("${IGC_BUILD__PROJ__fcl_dll}" PROPERTIES
+                      VERSION "${IGC_API_MAJOR_VERSION}.${IGC_API_MINOR_VERSION}.${IGC_API_PATCH_VERSION}"
+                      SOVERSION "${IGC_API_MAJOR_VERSION}")
 unset(_cpuSuffix)
 
 # Injecting external vISA project.
diff --git a/visa/iga/IGALibrary/CMakeLists.txt b/visa/iga/IGALibrary/CMakeLists.txt
index f4e9d34..10a97e5 100644
--- a/visa/iga/IGALibrary/CMakeLists.txt
+++ b/visa/iga/IGALibrary/CMakeLists.txt
@@ -329,6 +329,10 @@ if(ANDROID AND MEDIA_IGA)
 endif(ANDROID AND MEDIA_IGA)
 # target_link_libraries(IGA PRIVATE GEDLibrary)
 
+  set_target_properties(IGA_DLL PROPERTIES
+                        VERSION "${IGC_API_MAJOR_VERSION}.${IGC_API_MINOR_VERSION}.${IGC_API_PATCH_VERSION}"
+                        SOVERSION "${IGC_API_MAJOR_VERSION}")
+
   if(UNIX)
     INSTALL(TARGETS IGA_DLL
         RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/.   COMPONENT igc-core
